packages/vitest-plugin-storybook/src/global-setup.ts

```diff
@@ -0,0 +1,73 @@
+import net from 'node:net'
+import execa from 'execa'
+import { log } from './utils'
+
+import type { GlobalSetupContext } from 'vitest/node'
+
+let storybookProcess: execa.ExecaChildProcess | null = null
+
+const startStorybookIfNeeded = async () => {
+  const storybookScript = process.env.__STORYBOOK_SCRIPT__
+  const storybookUrl = process.env.__STORYBOOK_URL__ || ''
+
+  const url = new URL(storybookUrl)
+  const port = Number.parseInt(url.port, 10)
+
+  if (typeof port !== 'number' || Number.isNaN(port)) {
```

Comment:
When would it ever not be a typeof `number`? Seems like an unnecessary check.

Reply:
What if you run your Storybook server behind a proxy (e.g. as a route on your Next.js app)? There might not be a port on the URL at all. E.g. we use `https://storybook.dev-chromatic.com`.

Reply:
What do you think is the best experience for users? Would you set up both the Storybook URL as well as a port? Would port be only needed if you have a custom URL like that?

Reply:
>When would it ever not be a typeof number? Seems like an unnecessary check.
When people have URLs like the one you mentioned. If we have a port option, then we could throw an error so they pass the port as a plugin option